x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -247,33 +273,43 @@ export class AlertsClient {
       page,
       perPage,
       total,
-      data: data.map(({ id, attributes, updated_at, references }) =>
-        this.getAlertFromRaw(id, attributes, updated_at, references)
-      ),
+      data: data.map(({ id, attributes, updated_at, references }) => {
+        ensureAlertTypeIsAuthorized(attributes.alertTypeId, attributes.consumer);
```

Comment:
If the `authorizationFilter` does its job, this ensure isn't needed, but just incase some magical magic provided by the user in the `filter` or `search` fields manages to get around our filtering, we ensure no alert the user isn't authorized to read is actually returned.

I chose to make this an error case (rather than filter it out) as I want us to be explicit about unauthorized access and filtering will likely lead to buggy pagination behaviour which will be less likely to point us to the real issue which would be a potential security bug.